tests/Client.hs

```diff
@@ -60,12 +79,35 @@ data AnnotatedOrigin = AnnotatedOrigin
     aoOrigin :: M.Origin
   }
 
+-- | The Client store all configuration for the client.
+-- Notably it allows us to create a misbehaving client.
+-- Options with a distinct correct value are Maybe values, a Nothing value will
+-- assume the presumed correct behaviour is desired.
+data Client = Client
+  { -- | Changes the type of the attestation client data to whatever specified,
+    -- defaults to "webauthn.create"
+    cAttestationType :: Maybe JS.DOMString,
+    -- | Changes the type of the assertion client data to whatever specified
+    -- defaults to "webauthn.get"
+    cAssertionType :: Maybe JS.DOMString,
```

Comment:
Should also be an enum here. We want to have quickcheck generate such clients, and we don't want it to generate hundreds of strings for this scenario where there's really only 2: A valid string or an invalid one.